azkaban-hadoop-security-plugin/src/main/java/azkaban/security/HadoopSecurityManager_H_2_0.java

```diff
@@ -804,6 +771,77 @@ private void getToken(String userToProxy) throws InterruptedException,
 
   }
 
+  /**
+   * Prepare token file.
+   *  Writes credentials to a token file and sets permissions such that only the user being proxied has read access
+   *  to the file
+   *
+   * @param user user to be proxied
+   * @param credentials Credentials to be written to file
+   * @param tokenFile file to be written
+   * @param logger logger to use
+   * @throws IOException
+   */
+  private void prepareTokenFile(final String user,
+                                final Credentials credentials,
+                                final File tokenFile,
+                                final Logger logger) throws IOException {
+    writeCredentialsToFile(credentials, tokenFile, logger);
+    try {
+      assignPermissions(user, tokenFile, logger);
+    } catch (IOException e) {
+      // On any error managing token file. delete the file
+      tokenFile.delete();
+      throw e;
+    }
+  }
+
+  private void writeCredentialsToFile(Credentials credentials, File tokenFile, Logger logger) throws IOException {
+    FileOutputStream fos = null;
+    DataOutputStream dos = null;
+    try {
+      fos = new FileOutputStream(tokenFile);
+      dos = new DataOutputStream(fos);
+      credentials.writeTokenStorageToStream(dos);
+    } finally {
+      if (dos != null) {
+        try {
+          dos.close();
+        } catch (Throwable t) {
+          // best effort
+          logger.error("encountered exception while closing DataOutputStream of the tokenFile", t);
+        }
+      }
+      if (fos != null) {
+        fos.close();
+      }
+    }
+  }
+
+  /**
+   * Uses execute-as-user binary to reassign file permissions to be readable only by that user.
+   *
+   * Step 1. Set file permissions to 460. Readable to self and readable / writable azkaban group
+   * Step 2. Set user as owner of file.
+   *
+   * @param user user to be proxied
+   * @param tokenFile file to be written
+   * @param logger logger to use
+   */
+  private void assignPermissions(String user, File tokenFile, Logger logger) throws IOException {
+    final String changePermissionsCommand = String.format("chmod 460 %s", tokenFile.getAbsolutePath());
+    int result = executeAsUser.execute(System.getProperty("user.name"), changePermissionsCommand);
+    if (result != 0) {
+      throw new IOException("Unable to modify permissions. User: " + user);
+    }
+
+    final String changeOwnershipCommand = String.format("chown %s:azkaban %s", user, tokenFile.getAbsolutePath());
```

Comment:
This assumes that the user running this process is "azkaban". It's probably a safe assumption. It would be better if this is documented somewhere. 

Should 'azkaban' be a constant?